lib/board.rb

```diff
@@ -0,0 +1,55 @@
+require 'pry'
+
+class Board 
+  attr_accessor :cells
+
+  def initialize(cells = Array.new(9, " "))
+    @cells = cells
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " "
+    puts " "
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    puts " "
+    puts " "
+  end
+
+
+  def update(position, player)
+    @cells[position.to_i - 1] = player.token
+  end
+
+  def position(position)
+    @cells[position.to_i - 1]
+  end
+
+  def taken?(position)
+    !(@cells[position.to_i - 1].nil? || @cells[position.to_i - 1] == " ")
+  end
+
+  def valid_move?(position)
+    if (taken?(position.to_i) == false) && (position.to_i.between?(1,9) == true)
+    true
+    else (taken?(position.to_i) == true) || (position.to_i.between?(1,9) == false)
+    false
+    end
```

Comment:
Look out for patterns like
```
if #some condition
  return true
else
  return false
end
```
It is generally preferable to rely on the implicit return value of the boolean expression you are evaluating rather than returning `true` or `false`. Similarly, you should usually avoid equality comparisons with those boolean literals. In this case, you could rewrite your method as
```
def valid_move?(position)
  !taken?(position.to_i) && position.to_i.between?(1,9)
end
```